### PR TITLE
feat: add node-zendesk attachment response types

### PR DESCRIPTION
**Pull request by 9renpoto**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/node-zendesk/node-zendesk-tests.ts [hunk 1]
```diff
@@ -190,3 +191,25 @@ client.userfields.create({
...
+    function (err, req, result) {
+      if (err) {
+        console.log(err);
+        return;
+      }
+      console.log(JSON.stringify(result, null, 2));
+    });
+
+;(async () => {
+    const r1: Attachments.ShowResponseModel = await client.attachments.show(1);
```
Comment (9renpoto) [inline types/node-zendesk/node-zendesk-tests.ts:210]: https://developer.zendesk.com/rest_api/docs/support/attachments#example-response

File: types/node-zendesk/node-zendesk-tests.ts [hunk 2]
```diff
@@ -190,3 +191,25 @@ client.userfields.create({
...
+    {
+      filename: 'busey.gif'
+    },
+    function (err, req, result) {
+      if (err) {
+        console.log(err);
+        return;
+      }
+      console.log(JSON.stringify(result, null, 2));
+    });
```
Comment (9renpoto) [inline types/node-zendesk/node-zendesk-tests.ts:207]: https://github.com/blakmatrix/node-zendesk/blob/b69e783756511583f12b855ded9ad6c792a1b80d/examples/upload-attachment.js#L13-L24

File: types/node-zendesk/index.d.ts [hunk 3]
```diff
@@ -81,36 +81,54 @@ export namespace Attachments {
...
+     */
+    interface Attachment extends PersistableModel {
+        readonly content_type?: string;
+        readonly content_url?: string;
+        readonly deleted?: boolean;
+        readonly file_name?: string;
+        readonly inline?: boolean;
+        readonly mapped_content_url?: string;
+        readonly size?: number;
+        readonly url?: string;
```
Comment (9renpoto) [inline types/node-zendesk/index.d.ts:115]: https://developer.zendesk.com/rest_api/docs/support/attachments#json-format

**Review by tkiryu — APPROVED** [review]

**Review by tkiryu — APPROVED** [review]

**Comment by 9renpoto** [comment]
Ready to merge

**Pull request merged**